File: frontend/src/types/stats.ts
export interface StatsOverviewResp {
  total_practiced: number
  total_correct: number
  accuracy: number
  total_exams: number
  avg_exam_score: number
  study_days: number
  wrong_count: number
}

export interface DailyStatsResp {
  date: string
  total_count: number
  correct_count: number
  incorrect_count: number
}

export interface SubjectProgressResp {
  subject_id: number
  subject_name: string
  total_count: number
  correct_count: number
  accuracy: number
}

export interface CalendarStatsResp {
  date: string
  total_count: number
  correct_count: number
  incorrect_count: number
  duration: number
}

export interface ChapterProgressResp {
  chapter_id: number
  chapter_name: string
  sub_subject_id: number
  total_count: number
  correct_count: number
  accuracy: number
}

File: backend/internal/dto/stats.go
package dto

type StatsOverviewResp struct {
	TotalPracticed int64   `json:"total_practiced"`
	TotalCorrect   int64   `json:"total_correct"`
	Accuracy       float64 `json:"accuracy"`
	TotalExams     int64   `json:"total_exams"`
	AvgExamScore   float64 `json:"avg_exam_score"`
	StudyDays      int64   `json:"study_days"`
	WrongCount     int64   `json:"wrong_count"`
}

type DailyStatsResp struct {
	Date           string `json:"date"`
	TotalCount     int64  `json:"total_count"`
	CorrectCount   int64  `json:"correct_count"`
	IncorrectCount int64  `json:"incorrect_count"`
}

type SubjectProgressResp struct {
	SubjectID    uint    `json:"subject_id"`
	SubjectName  string  `json:"subject_name"`
	TotalCount   int64   `json:"total_count"`
	CorrectCount int64   `json:"correct_count"`
	Accuracy     float64 `json:"accuracy"`
}

type CalendarStatsResp struct {
	Date           string `json:"date"`
	TotalCount     int64  `json:"total_count"`
	CorrectCount   int64  `json:"correct_count"`
	IncorrectCount int64  `json:"incorrect_count"`
	Duration       int64  `json:"duration"`
}

type ChapterProgressResp struct {
	ChapterID    uint    `json:"chapter_id"`
	ChapterName  string  `json:"chapter_name"`
	SubSubjectID uint    `json:"sub_subject_id"`
	TotalCount   int64   `json:"total_count"`
	CorrectCount int64   `json:"correct_count"`
	Accuracy     float64 `json:"accuracy"`
}
